src/server/dfly_main.cc

```diff
@@ -161,35 +161,39 @@ struct VersionMonitor {
   }
 
  private:
-  void RunTask(SSL_CTX* ssl_ctx);
+  struct ssl_deleter {
+    void operator()(SSL_CTX* ssl) {
+      if (ssl) {
+        free(ssl);
+      }
+    }
+  };
+  using ssl_ptr = std::unique_ptr<SSL_CTX, ssl_deleter>;
+  void RunTask(ssl_ptr);
 };
 
 void VersionMonitor::Run(ProactorPool* proactor_pool) {
   // Avoid running dev environments.
-  bool is_dev_env = false;
-  const char* env_var = getenv("DFLY_DEV_ENV");
-  if (env_var) {
+  if (getenv("DFLY_DEV_ENV")) {
     LOG(WARNING) << "Running in dev environment (DFLY_DEV_ENV is set) - version monitoring is "
                     "disabled";
-    is_dev_env = true;
```

Comment:
you removed the is_dev_env, so now the check is running for dev env?

Reply:
Yep I moved it but I also though I added  a return statement above :/  Fixed.